Dockerfile

```diff
@@ -201,55 +226,74 @@ RUN /usr/bin/meson --prefix=${PREFIX} build \
         -Dresize-pool=false \
         -Dwcap-decode=false \
         -Dtest-junit-xml=false && \
-    ninja -C build -j8 install
-RUN echo 'weston:' `git --git-dir=/work/vendor/weston/.git rev-parse --verify HEAD` >> /work/versions.txt
+    ninja -C build -j8 install && \
+    echo 'weston:' `git --git-dir=/work/vendor/weston/.git rev-parse --verify HEAD` >> /work/versions.txt
+
+WORKDIR /work/debuginfo
+RUN if [ -z "$SYSTEMDISTRO_DEBUG_BUILD" ] ; then \
+        echo "== Strip debug info: weston ==" && \
+        /work/debuginfo/gen_debuginfo.sh /work/debuginfo/weston.list /work/build; \
+    fi
 
 # Build PulseAudio
 COPY vendor/pulseaudio /work/vendor/pulseaudio
 WORKDIR /work/vendor/pulseaudio
 RUN /usr/bin/meson --prefix=${PREFIX} build \
-        --buildtype=${BUILDTYPE} \
+        --buildtype=${BUILDTYPE_NODEBUGSTRIP} \
         -Ddatabase=simple \
         -Dbluez5=false \
         -Dgsettings=disabled \
         -Dtests=false && \
-    ninja -C build -j8 install
-RUN echo 'pulseaudio:' `git --git-dir=/work/vendor/pulseaudio/.git rev-parse --verify HEAD` >> /work/versions.txt
+    ninja -C build -j8 install && \
+    echo 'pulseaudio:' `git --git-dir=/work/vendor/pulseaudio/.git rev-parse --verify HEAD` >> /work/versions.txt
 
 # Build mesa with the minimal options we need.
 COPY vendor/mesa /work/vendor/mesa
 WORKDIR /work/vendor/mesa
 RUN /usr/bin/meson --prefix=${PREFIX} build \
-        --buildtype=${BUILDTYPE} \
+        --buildtype=${BUILDTYPE_NODEBUGSTRIP} \
         -Dgallium-drivers=swrast,d3d12 \
         -Ddri-drivers= \
         -Dvulkan-drivers= \
         -Dllvm=disabled && \
-    ninja -C build -j8 install
-RUN echo 'mesa:' `git --git-dir=/work/vendor/mesa/.git rev-parse --verify HEAD` >> /work/versions.txt
+    ninja -C build -j8 install && \
+    echo 'mesa:' `git --git-dir=/work/vendor/mesa/.git rev-parse --verify HEAD` >> /work/versions.txt
 
+# Build WSLGd Daemon
 ENV CC=/usr/bin/clang
 ENV CXX=/usr/bin/clang++
 
-# Build WSLGd Daemon
 COPY WSLGd /work/WSLGd
 WORKDIR /work/WSLGd
 RUN /usr/bin/meson --prefix=${PREFIX} build \
         --buildtype=${BUILDTYPE} && \
     ninja -C build -j8 install
 
+WORKDIR /work/debuginfo
+RUN if [ -z "$SYSTEMDISTRO_DEBUG_BUILD" ] ; then \
+        echo "== Strip debug info: WSLGd ==" && \
+        /work/debuginfo/gen_debuginfo.sh /work/debuginfo/WSLGd.list /work/build; \
+    fi
+
+# Gather debuginfo to a tar file
+WORKDIR /work/debuginfo
+RUN if [ -z "$SYSTEMDISTRO_DEBUG_BUILD" ] ; then \
+        echo "== Compress debug info: /work/debuginfo/system-distro-debuginfo.tar.gz ==" && \
+        tar -C /work/build/debuginfo -czf system-distro-debuginfo.tar.gz ./ ; \
```

Comment:
@viniciusjarina, should I append architecture suffix, like system-distro-debuginfo_x64.tar.gz here ? or better do rename-copy when publishing artifact ?

Reply:
Would be nice to have the arch sufix, then later if we need to download the artifact we can distinguish between ARM64 and x64